addons/io/org.openhab.io.neeo/src/main/java/org/openhab/io/neeo/NeeoService.java

```diff
@@ -0,0 +1,691 @@
+        Objects.requireNonNull(channelTypeRegistry, "channelTypeRegistry cannot be null");
+        contextLock.lock();
+        try {
+            this.channelTypeRegistry = channelTypeRegistry;
+        } finally {
+            contextLock.unlock();
+        }
+    }
+
+    /**
+     * Unset channel type registry.
+     *
+     * @param channelTypeRegistry the channel type registry (ignored)
+     */
+    public void unsetChannelTypeRegistry(ChannelTypeRegistry channelTypeRegistry) {
+        contextLock.lock();
+        try {
+            this.channelTypeRegistry = null;
+        } finally {
+            contextLock.unlock();
+        }
+    }
+
+    /**
+     * Sets the MDNS client.
+     *
+     * @param mdnsClient the non-null MDNS client
+     */
+    @Reference(cardinality = ReferenceCardinality.MANDATORY, name = "MDNSClient", policy = ReferencePolicy.STATIC, unbind = "unsetMDNSClient")
+    public void setMDNSClient(MDNSClient mdnsClient) {
+        Objects.requireNonNull(mdnsClient, "mdnsClient cannot be null");
+        contextLock.lock();
+        try {
+            this.mdnsClient = mdnsClient;
+        } finally {
+            contextLock.unlock();
+        }
+    }
+
+    /**
+     * Unset MDNS client.
+     *
+     * @param mdnsClient the mdns client (ignored)
+     */
+    public void unsetMDNSClient(MDNSClient mdnsClient) {
+        contextLock.lock();
+        try {
+            this.mdnsClient = null;
+        } finally {
+            contextLock.unlock();
+        }
+    }
+
+    /**
+     * Sets the event publisher.
+     *
+     * @param eventPublisher the new event publisher
+     */
+    @Reference(cardinality = ReferenceCardinality.MANDATORY, name = "EventPublisher", policy = ReferencePolicy.STATIC, unbind = "unsetEventPublisher")
+    public void setEventPublisher(EventPublisher eventPublisher) {
+        Objects.requireNonNull(eventPublisher, "eventPublisher cannot be null");
+        contextLock.lock();
+        try {
+            this.eventPublisher = eventPublisher;
+        } finally {
+            contextLock.unlock();
+        }
+    }
+
+    /**
+     * Unset event publisher.
+     *
+     * @param eventPublisher the event publisher (ignored)
+     */
+    public void unsetEventPublisher(EventPublisher eventPublisher) {
+        contextLock.lock();
+        try {
+            this.eventPublisher = null;
+        } finally {
+            contextLock.unlock();
+        }
+    }
+
+    /**
+     * Sets the network address service
+     *
+     * @param networkAddressService the network address service
+     */
+    @Reference(cardinality = ReferenceCardinality.MANDATORY, name = "NetworkAddressService", policy = ReferencePolicy.STATIC, unbind = "unsetNetworkAddressService")
+    public void setNetworkAddressService(NetworkAddressService networkAddressService) {
+        Objects.requireNonNull(networkAddressService, "networkAddressService cannot be null");
+        contextLock.lock();
```

Comment:
I'm not sure I really need this anymore - had something to do with deactivation (vaguely remember the servlet took time to close down and I had some threading issue with the servlet shutdown and the unsets).  But since that time, I rewrote using the servercontext (which will still have references).  Let me think this one through again - I may just eliminate the lock fully